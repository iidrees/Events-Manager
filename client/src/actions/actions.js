import axios from 'axios';
import { history } from '../routes';

import { 
  SIGN_UP, SIGNED_UP, 
  SIGN_UP_FAIL, SIGN_IN, 
  SIGNED_IN, SIGN_IN_FAIL 
} from './types';

/**
 * Axios will help make POST request to signup user
 * @export {function}
 * @param {any} userData
 * @returns {JSON} userData
 */
export const userSignup = (userData) => {
  return (dispatch) => {
    dispatch({ type: 'SIGN_UP' });
    axios({
      method: 'post',
      url: 'http://localhost:5050/api/v1/users',
      data: userData,
      withCredentials: true,
    })
      .then((response) => {
        dispatch({ type: SIGNED_UP, payload: response.data });
        history.push('/getEvents');
      })
      .catch((err) => {
        dispatch({ type: 'SIGN_UP_FAIL', payload: err });
        history.push('/');
      });
  };
};

export const userSignin = (userData) => {
  return (dispatch) => {
    dispatch({ type: 'SIGN_IN' });
    axios({
      method: 'post',
      url: 'http://localhost:5050/api/v1/users/login',
      data: userData,
      withCredentials: true,
    })
      .then((response) => {
        dispatch({ type: SIGNED_IN, payload: response.data });
        history.push('/login');
      })
      .catch((err) => {
        console.log(err.response.data);
        dispatch({ type: 'SIGNED_IN_FAIL', payload: err });
      // history.push('/login);
      });
  };
};

/**
 * @export {function}
 * @param {any} userData
 * @returns {JSON} userData
 */
/* export function userSigninRequest(userData) {
  return dispatch => axios.post('/api/v1/users/login', userData);
}
 */
