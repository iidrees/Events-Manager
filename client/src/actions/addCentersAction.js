import axios from 'axios';

import {
  ADD_CENTER,
  ADD_CENTER_FAIL
} from './types'

/* eslint-disable */
export const addCenter = (centerData) => {
  console.log('centerData from actions', centerData);
  return (dispatch) => {
    axios({
      method: 'post',
      url: '/api/v1/centers',
      data: centerData,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((res) => {
      console.log('res from actions addcenter',  res);
      dispatch(({ type: ADD_CENTER, res }))
    })
    .catch((err) => {
      console.log('err from actions addcenters', err);
      dispatch({ type: ADD_CENTER_FAIL, err });
    })
  }
}