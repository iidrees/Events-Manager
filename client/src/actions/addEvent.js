/* Import dependencies and Modules */

import axios from 'axios';
import { history } from '../routes';

import {
  ADD_EVENT,
  ADD_EVENT_FAIL,
  ADD_IMG_FAIL,
  GET_EVENT,
  ADD_EVENT_LOAD
} from './types';

/**
 * Axios will help make POST request to add event
 * @export {function} addEvent function action
 * @param {eventData} eventData object sent to the server
 * @param {centerId}  centerId {centerId}
 * @param {imgUrl} imgUrl {the cloudinary url}
 * @returns {JSON} JSON data containing events
 */
export const addEvent = (eventData, centerId, imgUrl) => {
  return dispatch => {
    return axios({
      method: 'post',
      url: `/api/v1/events/${centerId}`,
      data: {
        title: eventData.title,
        startDate: eventData.startDate,
        endDate: eventData.endDate,
        center: eventData.center,
        description: eventData.description,
        imgUrl: imgUrl
      },
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: ADD_EVENT, event: response.data });
      })
      .catch(err => {
        dispatch({ type: ADD_EVENT_FAIL, error: err.response.data });
      });
  };
};

/**
 * @export {function} imageUpload function
 * @param {eventData} eventData object sent to the server
 * @param {centerId}  centerId {centerId}
 * @returns {URL} URL link returned is used as a parameter
 */
export const imageUpload = (eventData, centerId) => {
  let formData = new FormData();

  formData.append('file', eventData.imgFile);
  formData.append('upload_preset', process.env.UPLOAD_PRESET);
  return dispatch => {
    dispatch({ type: ADD_EVENT_LOAD });
    axios({
      method: 'post',
      url: process.env.CLOUDINARY_URL,
      data: formData,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
      .then(response => {
        let imageURL = response.data.secure_url;
        return dispatch(addEvent(eventData, centerId, imageURL));
      })
      .catch(err => {
        dispatch({
          type: ADD_IMG_FAIL,
          error: 'Image upload failed'
        });
      });
  };
};
