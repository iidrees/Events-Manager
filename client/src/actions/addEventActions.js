/* Import dependencies and Modules */

import axios from 'axios';
import { history } from '../routes';

import { 
  ADD_EVENT, 
  ADD_EVENT_FAIL, 
  ADD_IMG_FAIL,
  GET_EVENT
} from './types';


/**
 * Axios will help make POST request to add event
 * @export {function} addEvent function action
 * @param {eventData} eventData object sent to the server
 * @param {index}  index {centerId}
 * @param {imgUrl} imgUrl {the cloudinary url}
 * @returns {JSON} JSON data containing events 
 */
export const addEvent = (eventData, index, imgUrl ) => {
  
  return (dispatch) => {
    return axios({
      method: 'post',
      url: `/api/v1/events/${index}`,
      data: { 
        title: eventData.title,
        date: eventData.date,
        time: eventData.time,
        center: eventData.center,
        description: eventData.description,
        type: eventData.type,
        imgUrl: imgUrl 
      },
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((response) => {
      dispatch({ type: GET_EVENT, event: response.data.data})
    })
    .catch((err) => {
      dispatch({ type: ADD_EVENT_FAIL, error: err.response.data });
    })
  }
}


/**
 * @export {function} imageUpload function
 * @param {eventData} eventData object sent to the server
 * @param {index}  index {centerId}
 * @returns {URL} URL link returned is used as a parameter
 */
export const imageUpload = (eventData,index) => {
  
  let formData = new FormData();
  
  formData.append('file', eventData.imgFile)
  formData.append('upload_preset', process.env.UPLOAD_PRESET)
  return (dispatch) => {
    axios({
      method: 'post',
      url: process.env.CLOUDINARY_URL,
      data: formData,
      headers: {
        'Content-Type':'application/x-www-form-urlencoded'
      }
    })
    .then(response => {
      let imageURL = response.data.secure_url;
      console.log('cloudinary is done here>>>>>>')
      return dispatch(addEvent(eventData,index,imageURL))
    })
    .catch(err => {

      dispatch({
        type: ADD_IMG_FAIL, 
        error: 'Image upload failed'
      })
    })
  }
} 