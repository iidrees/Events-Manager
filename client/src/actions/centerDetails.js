import axios from 'axios';
import { history } from '../routes';

import {
  GET_CENTER,
  GET_CENTER_FAIL,
  CANCEL_EVENT,
  CANCEL_EVENT_FAIL
} from './types';

/**
 *  @param {centerId} centerId the center's index
 * @param {page} page - page number
 * @returns {JSON} object
 */
export const centerDetails = (centerId, page) => {
  return dispatch => {
    return axios({
      method: 'GET',
      url: `/api/v1/centers/${centerId}?page=${page}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: GET_CENTER, response: response.data });
      })
      .catch(err => {
        dispatch({ type: GET_CENTER_FAIL, error: err.response.data });
      });
  };
};

/**
 * @param {index} eventId - eventId
 * @returns {void}
 */
export const cancelEvent = eventId => {
  return dispatch => {
    return axios({
      method: 'PUT',
      url: `/api/v1/cancelEvent/${eventId}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: CANCEL_EVENT, response: response.data });
      })
      .catch(err => {
        dispatch({ type: CANCEL_EVENT_FAIL, error: err.response.data });
      });
  };
};
