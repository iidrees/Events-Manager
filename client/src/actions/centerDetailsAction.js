import axios from 'axios';
import { history } from '../routes';

import { 
  GET_CENTER,
  GET_CENTER_FAIL

  } from './types';

/**
 * 
 * 
 * @param {any} index the center's index
 * @returns {JSON} object
 */
export const centerDetails = (index,page) => { // eslint-disable-line
  return (dispatch) => {
   return axios({
      method: 'GET',
      url: `/api/v1/centers/${index}?page=${page}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
    .then((response) => {
      console.log('the details actions>>>', response.data)
      dispatch({ type: GET_CENTER, response: response.data })
    })
    .catch((err) => {
      dispatch({ type: GET_CENTER_FAIL, error: err.response.data});
    })
  }
}