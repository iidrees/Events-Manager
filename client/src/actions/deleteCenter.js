import axios from 'axios';
import { history } from '../routes';

import { DELETE_CENTER, DELETE_CENTER_FAIL } from './types';

/**
 *
 *
 * @param {centerId} centerId - centerId
 * @returns {void} -
 */
const deleteCenter = centerId => {
  return dispatch => {
    return axios({
      method: 'DELETE',
      url: `/api/v1/centers/${centerId}`,
      headers: {
        'x-access-token': localStorage.getItem('x-access-token')
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({ type: DELETE_CENTER, centerDeleted: response.data });
      })
      .catch(err => {
        dispatch({ type: DELETE_CENTER_FAIL, error: err.response.data });
      });
  };
};

export default deleteCenter;
