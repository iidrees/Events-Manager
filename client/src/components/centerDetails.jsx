import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { deleteCenter } from '../actions/deleteCenterAction';
import {centerDetails } from '../actions/centerDetailsAction';

/**
 * 
 * 
 * @class CenterDetails
 * @extends {React.Component}
 */
class CenterDetails extends React.Component {
/**
 * 
 * 
 * @returns {JSON} JSON
 * @memberof CenterDetails
 */
componentWillMount() {
    const { dispatch } = this.props;
    console.log('This is details dispatch >>>>')
    return dispatch(centerDetails(this.props.match.params.id));
  }
/**
 * 
 * @returns {null} null
 * @param {event} event -
 * @memberof CenterDetails
 */
onDelete = (event) => {
    event.preventDefault();
    let { center } = this.props;
    const { dispatch } = this.props;
    return dispatch(deleteCenter(center.id));
  }
/**
 * 
 * 
 * @returns {JSX} JSX
 * @memberof CenterDetails
 */
render () {

    const { center, user } = this.props;

    return (
      <div>
        <div>
          {
            (!user.authenticated) &&
            <Redirect to='/signin' push />
          }
        </div>
      <div className="container">
        <div className="row">
          <div className="container" id="center-details-header">{/* START PAGE HEADER */}
            <div className="row">
              <div className="col-sm-12">              
                  <h1 className="text-center head-1">Event center details </h1>
                    <p className="head-para text-center">
                    The {center.name}
                    </p>
              </div>
            </div>
            <hr/>
          </div>{/* END PAGE-HEADER */}

          <div className="container" id="center-details-slider"> {/* <!-- START OF EVENT CENTER DETAILS --> */}
            <div className="row">
              <div className="col-md-6">{/* <!--  START OF SLIDER COLUMN --> */}
                  <div id="carousel-landing" className="carousel slide" data-ride="carousel">{/* <!-- START TAG FOR CAROUSEL SLIDER   --> */}
                    <ol className="carousel-indicators">
                      <li data-target="#carousel-landing" data-slide-to="0" className="active"></li>
                      <li data-target="#carousel-landing" data-slide-to="1"></li>
                      <li data-target="#carousel-landing" data-slide-to="2"></li>
                    </ol>
                    <div className="carousel-inner" role="listbox">
                      <div className="carousel-item active">
                        <img id="img-landing1" className="d-block img-fluid" src="https://static.pexels.com/photos/3865/lights-night-firework-new-year-s-eve.jpg" style={{width: '100%', height: '65%'}} alt="First slide"/>
                      </div>
                      <div className="carousel-item">
                        <img id="img-landing2" className="d-block img-fluid" src="https://static.pexels.com/photos/274192/pexels-photo-274192.jpeg"  style={{width: '100%', height: '65%'}} alt="Second slide"/>
                      </div>
                      <div className="carousel-item">
                        <img id="img-landing3" className="d-block img-fluid" src="https://static.pexels.com/photos/169190/pexels-photo-169190.jpeg"  style={{width: '100%', height: '65%'}} alt="Third slide"/>
                      </div>
                      <div className="carousel-item">
                          <img id="img-landing4" className="d-block img-fluid" src="https://static.pexels.com/photos/169211/pexels-photo-169211.jpeg"  style={{width: '100%', height: '65%'}} alt="Third slide"/>
                      </div>
                    </div>
                    <a className="carousel-control-prev" href="#carousel-landing" role="button" data-slide="prev">
                      <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                      <span className="sr-only">Previous</span>
                    </a>
                    <a className="carousel-control-next" href="#carousel-landing" role="button" data-slide="next">
                      <span className="carousel-control-next-icon" aria-hidden="true"></span>
                      <span className="sr-only">Next</span>
                    </a>
                </div>{/* <!--  END Carousel slider Tag --> */}
              </div>{/* <!-- END OF SLIDER COLUMN --> */}
              
              <div className="col-md-6">
                <p className="center-details-para">
                  <span className="center-details-span">Center:</span>
                  The {center.name}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Description:</span> <br/>
                  {center.description}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Location:</span>
                  {center.location}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Capacity:</span>
                  {center.capacity}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span" >Owner:</span>
                  {center.owner}
                </p>
                <p className="center-details-para">
                </p>
                <Link className="btn btn-primary" to={`/editcenter/${center.id}`} role="button">Edit Center Details</Link>
                {/* <button className="btn btn-danger" onClick={this.onDelete}  role="button">DELETE CENTER</button> */}
                <button type="button" className="btn btn-danger" data-toggle="modal" data-target="#deleteCenter">
                    DELETE
                  </button>
                {/* <!-- Modal --> */}
                <div className="modal fade" id="deleteCenter" tabIndex="-1" role="dialog" aria-labelledby="deleteCenterLabel" aria-hidden="true">
                    <div className="modal-dialog" role="document">
                      <div className="modal-content">
                        <div className="modal-header">
                          <h5 className="modal-title" id="deleteCenterLabel">ARE YOU SURE YOU WANT TO DELETE THIS CENTER?</h5>
                          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                          </button>
                        </div>
                        <div className="modal-body">
                        The {center.name}
                        </div>
                        <div className="modal-footer">
                          <button type="button" className="btn btn-primary" data-dismiss="modal">No</button>
                          <button className="btn btn-warning" onClick={this.onDelete} type="button">Delete Center</button>
                          {/* <button type="button" className="btn btn-primary">Save changes</button> */}
                        </div>
                      </div>
                    </div>
                  </div>
              </div>
            </div>
            <hr/>
          </div>{/* <!-- END OF EVENT CENTER DETAILS --> */}

          <div className="container" id="event-holding-header">
            <div className="row">
              <div className="col-sm-12">              
                <h4 className="text-center" id="head-holding">
                  Events At This Event Center
                </h4>
              </div>
            </div>
            <hr/>
          </div>

          <div className="container" id="event-holding">
            <div className="row">
           {/* {center.events.map((event) => {
              return (

              <div className="col-sm-4" key={event.id}>
              <div className="card-deck cont-body" id="card-body">
                <div className="card" id="card1" style={{height: "10rem"}}>
                  <img className="card-img-top" src="https://static.pexels.com/photos/122250/pexels-photo-122250.jpeg" alt="Card image cap"/>
                  <div className="card-block" style={{borderBottom: "solid grey 0.5px"}}>
                    <h5 className="card-title even-font">
                      <span >Event:</span>
                      {event.title}
                    </h5>
                    <p className="card-title date-font">
                        <span>DATE:</span>
                      {event.date}</p>
                    <p className="card-text">
                        <span>tags:</span>
                      <small className="text-muted">{event.type}</small></p>
                  </div>              
                </div>
              </div>
            </div>
              )
            })}  */}

            </div>
          </div>
          
        </div>
      </div>
      </div>
    )
  }

}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    center: state.centerDetailsReducer,
    user: state.userReducer
    
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CenterDetails);