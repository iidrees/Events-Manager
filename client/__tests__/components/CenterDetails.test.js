import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import sinon from 'sinon';

import { events } from '../actions/mocks/getEventsMock';
import { centerDetails, cancelEvent } from '../../src/actions/centerDetails';
import deleteCenter from '../../src/actions/deleteCenter';
import { DetailEvent, CenterDetails } from '../../src/components/CenterDetails';

describe('DetailEvent component', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error');
    console.error.mockImplementation(() => {}); // eslint-disable-line
  });

  afterEach(() => {
    console.error.mockRestore(); // eslint-disable-line
  });
  const locations = [];
  const history = {
    push(location) {
      locations.push(location);
    }
  };
  let props = {
    authenticated: true,
    user: {},
    center: {
      id: 1,
      name: 'The Amity Center',
      location: 'Anthony Lagos',
      address: 'No 2, Lagos way, Lagos',
      owner: 'Idrees-kun',
      capacity: 23333,
      description: 'This is a center that is a state of the art blah blah blah',
      imgUrl: 'https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg'
    },
    events,
    match: {
      params: {
        id: 1
      }
    },
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI',
    onDelete: jest.fn(),
    dispatch: jest.fn(),
    onCancel: jest.fn(),
    onChange: jest.fn()
  };
  let wrapper = shallow(
    <Router>
      <CenterDetails {...props} history={history} />
    </Router>
  );

  it('should render without throwing an error', () => {
    expect(
      mount(
        <Router>
          <CenterDetails {...props} history={history} />
        </Router>
      ).length
    ).toEqual(1);
  });
  expect(
    <Router>
      <CenterDetails {...props} history={history} />
    </Router>
  ).toMatchSnapshot();
});
