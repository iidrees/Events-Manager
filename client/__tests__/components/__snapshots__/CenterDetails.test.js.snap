// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<MemoryRouter>
  <CenterDetails
    authenticated={true}
    center={
      Object {
        "address": "No 2, Lagos way, Lagos",
        "capacity": 23333,
        "description": "This is a center that is a state of the art blah blah blah",
        "id": 1,
        "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
        "location": "Anthony Lagos",
        "name": "The Amity Center",
        "owner": "Idrees-kun",
      }
    }
    dispatch={[MockFunction]}
    events={
      Array [
        Object {
          "center": "The Multi-purpose",
          "date": "21-03-2019",
          "description": "This is going to be the weirdest parry on the month of March",
          "id": 1,
          "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
          "time": "10:00 PM",
          "title": "The Amity party",
          "type": "public",
        },
        Object {
          "center": "The Multi-purpose",
          "date": "21-03-2019",
          "description": "This is going to be the weirdest parry on the month of March",
          "id": 2,
          "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
          "time": "10:00 PM",
          "title": "The Amity party",
          "type": "public",
        },
        Object {
          "center": "The Multi-purpose",
          "date": "21-03-2019",
          "description": "This is going to be the weirdest parry on the month of March",
          "id": 3,
          "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
          "time": "10:00 PM",
          "title": "The Amity party",
          "type": "public",
        },
        Object {
          "center": "The Multi-purpose",
          "date": "21-03-2019",
          "description": "This is going to be the weirdest parry on the month of March",
          "id": 4,
          "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
          "time": "10:00 PM",
          "title": "The Amity party",
          "type": "public",
        },
        Object {
          "center": "The Multi-purpose",
          "date": "21-03-2019",
          "description": "This is going to be the weirdest parry on the month of March",
          "id": 5,
          "imgUrl": "https://static.pexels.com/photos/169193/pexels-photo-169193.jpeg",
          "time": "10:00 PM",
          "title": "The Amity party",
          "type": "public",
        },
      ]
    }
    history={
      Object {
        "push": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    onCancel={[MockFunction]}
    onChange={[MockFunction]}
    onDelete={[MockFunction]}
    token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoZW50aWNhdGVkIjp0cnVlLCJpZCI6MSwiYWRtaW4iOnRydWUsInJvbGUiOiJzdXBlckFkbWluIiwiaXNTdXBlckFkbWluIjp0cnVlLCJuYW1lIjoiSWRyZWVzIiwiaWF0IjoxNTI3NTk5OTkzLCJleHAiOjE1Mjc2MzU5OTN9.UAzE5IOIN3RVO7E4JCOnwtzpp5vberbZn8jOflEf1JI"
    user={Object {}}
  />
</MemoryRouter>
`;
